mm2src/coins/nft/storage/wasm/wasm_storage.rs

```diff
@@ -0,0 +1,235 @@
+
+    async fn is_initialized(&self, _chain: &Chain) -> MmResult<bool, Self::Error> { todo!() }
+
+    async fn get_tx_history(
+        &self,
+        _chains: Vec<Chain>,
+        _max: bool,
+        _limit: usize,
+        _page_number: Option<NonZeroUsize>,
+        _filters: Option<NftTxHistoryFilters>,
+    ) -> MmResult<NftsTransferHistoryList, Self::Error> {
+        todo!()
+    }
+
+    async fn add_txs_to_history<I>(&self, _chain: &Chain, _txs: I) -> MmResult<(), Self::Error>
+    where
+        I: IntoIterator<Item = NftTransferHistory> + Send + 'static,
+        I::IntoIter: Send,
+    {
+        todo!()
+    }
+
+    async fn get_last_block_number(&self, _chain: &Chain) -> MmResult<Option<u32>, Self::Error> { todo!() }
+
+    async fn get_txs_from_block(
+        &self,
+        _chain: &Chain,
+        _from_block: u32,
+    ) -> MmResult<Vec<NftTransferHistory>, Self::Error> {
+        todo!()
+    }
+
+    async fn get_txs_by_token_addr_id(
+        &self,
+        _chain: &Chain,
+        _token_address: String,
+        _token_id: BigDecimal,
+    ) -> MmResult<Vec<NftTransferHistory>, Self::Error> {
+        todo!()
+    }
+
+    async fn get_tx_by_tx_hash(
+        &self,
+        _chain: &Chain,
+        _transaction_hash: String,
+    ) -> MmResult<Option<NftTransferHistory>, Self::Error> {
+        todo!()
+    }
+
+    async fn update_tx_meta_by_hash(&self, _chain: &Chain, _tx: NftTransferHistory) -> MmResult<(), Self::Error> {
+        todo!()
+    }
+
+    async fn update_txs_meta_by_token_addr_id(&self, _chain: &Chain, _tx_meta: TxMeta) -> MmResult<(), Self::Error> {
+        todo!()
+    }
+
+    async fn get_txs_with_empty_meta(&self, _chain: &Chain) -> MmResult<Vec<NftTokenAddrId>, Self::Error> { todo!() }
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub(crate) struct NftListTable {
+    token_address: String,
+    token_id: BeBigUint,
+    chain: String,
+    amount: BeBigUint,
+    block_number: u32,
+    contract_type: ContractType,
+    details_json: Json,
+}
+
+impl NftListTable {
+    pub const CHAIN_TOKEN_ADD_TOKEN_ID_INDEX: &str = "chain_token_add_token_id_index";
+}
+
+impl TableSignature for NftListTable {
+    fn table_name() -> &'static str { "nft_list_cache_table" }
+
+    fn on_upgrade_needed(upgrader: &DbUpgrader, old_version: u32, new_version: u32) -> OnUpgradeResult<()> {
+        if let (0, 1) = (old_version, new_version) {
+            let table = upgrader.create_table(Self::table_name())?;
+            table.create_multi_index(
+                Self::CHAIN_TOKEN_ADD_TOKEN_ID_INDEX,
+                &["chain", "token_address", "token_id"],
+                true,
+            )?;
+            table.create_index("chain", false)?;
+        }
+        Ok(())
+    }
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub(crate) struct NftTxHistoryTable {
+    transaction_hash: String,
+    chain: String,
+    block_number: u32,
+    block_timestamp: u64,
```

Comment:
remember to use `BeBigUint` instead of `u64`.. I guess you know the reason why 👀

Reply:
:sweat_smile: yeah, I decided not to touch wasm storage, since Ive already added a lot of changes in `nft-cache-support-wasm` branch. well lets add `BeBigUint` in this branch too, anyway I have to resolve conflicts after merging this branch with native feature.